Modules/DataTypesExt/src/mitkDataStorageSelection.cpp

```
@@ -64,7 +64,27 @@ namespace mitk
 
   mitk::DataNode::Pointer DataStorageSelection::GetNode(unsigned int index) const
   {
-    return (index < m_Nodes.size())? m_Nodes.at(index): nullptr;
+    if (!m_Nodes.size())
+    {
+      return nullptr;
+    }
+    
+    mitk::DataNode::Pointer node = nullptr;
+    
+    try
+    {
+      node = m_Nodes.at(index);
+    }
+    catch(const std::out_of_range& exp)
+    {
+      return nullptr;
+    }
+    catch(...)
```

Такое лучше не делать (read "let it crash" pattern)